sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -1402,4 +1402,28 @@ class JoinSuite extends QueryTest with SharedSparkSession with AdaptiveSparkPlan
       assertJoin(sql, classOf[ShuffledHashJoinExec])
     }
   }
+
+  test("SPARK-36794: Ignore duplicated key when building relation for semi/anti hash join") {
```

Comment:
Looks like a good idea to ignore the duplicate keys, but this test passes with and without the patch, right? Seems  there isn't a way to show the difference?

Reply:
> but this test passes with and without the patch, right?

Yes. This unit test is added mostly to verify after adding this PR, the join still works as expected. And this PR is not fixing a regression, but just an improvement.

> Seems there isn't a way to show the difference?

Well in theory we can add more code to check the number of rows inside `HashedRelation`, and this should have a difference before/after this PR. However this would need more code change, e.g. introducing a new SQL metrics for number of rows inside `HashedRelation`, which looks like not unnecessary.